mne/time_frequency/tfr.py

```diff
@@ -1374,3 +1375,112 @@ def tfr_multitaper(inst, freqs, n_cycles, time_bandwidth=4.0,
+    if not all(isinstance(e, AverageTFR) for e in all_tfr):
+        raise ValueError("Not all the elements in list are AllTFR data")
+
+    # Copy channels to leave the original evoked datasets intact.
+    all_tfr = [e.copy() for e in all_tfr]
+
+    # Interpolates if necessary
+    if drop_bads:
+        for tfr in all_tfr:
+            tfr.drop_channels(tfr.info['bads'], copy=False)
+    else:
+        pass
+        # implement good channel subset here somehow??
+
+    equalize_channels(all_tfr)  # apply equalize_channels
+    # make grand_average object using combine_tfr
+    grand_average = combine_tfr(all_tfr)
+    # change the grand_average.nave to the number of Evokeds
+    grand_average.nave = len(all_tfr)
+    # change comment field
+    grand_average.comment = "Grand average (n = %d)" % grand_average.nave
+    return grand_average
+
+
+def combine_tfr(all_tfr, weights='nave'):
+    """Merge TFR data
+
+    Data should have the same channels and the same time instants.
+    Subtraction can be performed by passing negative weights (e.g., [1, -1]).
+
+    Parameters
+    ----------
+    all_tfr : list of TFR
+        The tfr datasets.
+    weights : list of float | str
+        The weights to apply to the data of each TFR instance.
+        Can also be ``'nave'`` to weight according to tfr.nave,
+        or ``"equal"`` to use equal weighting (each weighted as ``1/N``).
+
+    Returns
+    -------
+    tfr : AllTFR
+        The new TFR data.
+
+    Notes
+    -----
+    .. versionadded:: 0.10.0
+    """
+    tfr = all_tfr[0].copy()
+    if isinstance(weights, string_types):
+        if weights not in ('nave', 'equal'):
+            raise ValueError('Weights must be a list of float, or "nave" or '
+                             '"equal"')
+        if weights == 'nave':
+            weights = np.array([e.nave for e in all_tfr], float)
+            weights /= weights.sum()
+        else:  # == 'equal'
+            weights = [1. / len(all_tfr)] * len(all_tfr)
+    weights = np.array(weights, float)
+    if weights.ndim != 1 or weights.size != len(all_tfr):
+        raise ValueError('Weights must be the same size as all_tfr')
+
+    ch_names = tfr.ch_names
+    for e in all_tfr[1:]:
+        assert e.ch_names == ch_names, ValueError("%s and %s do not contain "
+                                                  "the same channels"
+                                                  % (tfr, e))
+        assert np.max(np.abs(e.times - tfr.times)) < 1e-7, \
+            ValueError("%s and %s do not contain the same time instants"
+                       % (tfr, e))
+
+    # use union of bad channels
+    bads = list(set(tfr.info['bads']).union(*(ev.info['bads']
+                                              for ev in all_tfr[1:])))
+    tfr.info['bads'] = bads
+
+    tfr.data = sum(w * e.data for w, e in zip(weights, all_tfr))
+    tfr.nave = max(int(1. / sum(w ** 2 / e.nave
+                                for w, e in zip(weights, all_tfr))), 1)
```

Comment:
combine_tfr or combine_evoked allows you to contrast data too using
negative weights.